### PR TITLE
Feature: Linux support for Application Search

### PR DESCRIPTION
**Pull request by ke1v**
Ported over the previous attempt to add Linux support into the v9 codebase

Currently it's broken due to the usage of `extract-file-icon`, which doesn't support Linux. I haven't modified FileImageGenerator since it is a core dependency. How should I approach integrating Linux support into the application?

### CODE REVIEW

**Comment by oliverschwendener** [comment]
Thanks alot for your contribution! Currently I'm working on the removal of `extract-file-icon` as it was not really stable on Windows and not working on Linux at all. After `extract-file-icon` was removed we can look at merging this PR. I'll keep you posted.

**Comment by oliverschwendener** [comment]
`extract-file-icon` is now removed. You can merge `main` back into your branch.

File: src/main/Core/ActionHandler/DefaultActionHandlers/OpenFilePathActionHandler.ts [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
     private async openFilePath(filePath: string): Promise<void> {
-        const errorMessage = await this.shell.openPath(filePath);
-
-        if (errorMessage) {
-            throw new Error(errorMessage);
+        // Linux has own way of opening files
+        if (extname(filePath) === ".desktop") {
+            // Don't await otherwise command hangs until app closes
+            this.commandlineUtility.executeCommand(`gtk-launch ${basename(filePath)}`);
+        } else {
```
Comment (oliverschwendener) [inline src/main/Core/ActionHandler/DefaultActionHandlers/OpenFilePathActionHandler.ts:24]: If we LInux has a special way to open applications, I suggest to create a custom action handler in the extension.

File: src/main/Core/FileSystemUtility/Contract/FileSystemUtility.ts [hunk 2]
```diff
@@ -4,6 +4,9 @@ export interface FileSystemUtility {
     pathExists(fileOrFolderPath: string): Promise<boolean>;
     readJsonFile<T>(filePath: string): Promise<T>;
     readJsonFileSync<T>(filePath: string): T;
+    readIniFile(filePath: string): Promise<Record<string, Record<string, string>>>;
+    readIniFileSync(filePath: string): Record<string, Record<string, string>>;
```
Comment (oliverschwendener) [inline src/main/Core/FileSystemUtility/Contract/FileSystemUtility.ts:8]: This is very specific to ini files, I suggest to create a separate utility for reading ini files.

**Comment by ke1v** [comment]
Force push was an accident. Misclicked and Github apparently does not have a confirmation screen when force-pulling from upstream. I can't seem to reopen the pull request after reverting the push either.

**Comment by oliverschwendener** [comment]
Can you open a new PR?

**Comment by oliverschwendener** [comment]
@ke1v By the way I started to add some of your changes into my main branch separately, so that we can open a PR which is purely for adding the linux support in the ApplicationSearch extension.

See here: https://github.com/oliverschwendener/ueli/commit/6d8a2d335ddd198fd3ac45e0ad6516896a5151d4
and here: https://github.com/oliverschwendener/ueli/commit/78ae5dba1c5afcb207b2e98648537bf27d9521cd

**Comment by oliverschwendener** [comment]
@ke1v And for the ini file parser I created a generic utility that can be used by the ApplicationSearch extension: #1077

Would you mind helping me with that?

**Pull request closed without merging**